src/batch/src/task/task_execution.rs

```diff
@@ -245,9 +252,20 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
             .extend(receivers.into_iter().map(Some));
         let failure = self.failure.clone();
         let task_id = self.task_id.clone();
+
+        // After we init the output receivers, it's must safe to schedule next stage -- send
+        // TaskStatus::Running here.
```

Comment:
By the way, please remove the unused states here:

https://github.com/singularity-data/risingwave/blob/946544c1200f7be02906537f2f2e1f23a3ca4e21/proto/task_service.proto#L20-L31

Reply:
I will do a refactor later to clean out all unused task status